src/pages/Search/components/Form.js

```diff
@@ -1,32 +1,54 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import { useHistory, useParams } from 'react-router-dom';
 import Button from '../../../components/button';
+import Loader from '../../../components/loader';
 import {
   FormContainer, InputGroup, Prefixed, Input,
 } from './Form.styles';
 
-const Form = () => {
+const Form = ({ loadData }) => {
   const { push } = useHistory();
   const { subreddit } = useParams();
   const [query, setQuery] = useState(subreddit);
+  const [loading, setLoading] = useState(false);
 
-  const handleSubmit = (e) => {
+  useEffect(() => {
+    const onLoad = async () => {
+      setLoading(true);
+      await loadData(query);
+      setLoading(false);
+    };
+    onLoad();
+  }, []);
```

Comment:
You could also add `subreddit` as a dependency. Then this would run for every change of `subreddit` and you wouldn't need this logic inside `handleSubmit`

Reply:
Or even better move this `useEffect` to `SearchPage`. Then this form is not connected to data loading at all. It only updates the URL